library/core/src/num/nonzero.rs

```diff
@@ -191,13 +191,12 @@ macro_rules! nonzero_leading_trailing_zeros {
                 /// Basic usage:
                 ///
                 /// ```
-                /// #![feature(nonzero_leading_trailing_zeros)]
                 #[doc = concat!("let n = std::num::", stringify!($Ty), "::new(", stringify!($LeadingTestExpr), ").unwrap();")]
                 ///
                 /// assert_eq!(n.leading_zeros(), 0);
                 /// ```
-                #[unstable(feature = "nonzero_leading_trailing_zeros", issue = "79143")]
-                #[rustc_const_unstable(feature = "nonzero_leading_trailing_zeros", issue = "79143")]
+                #[stable(feature = "nonzero_leading_trailing_zeros", since = "1.53.0")]
+                #[rustc_const_stable(feature = "nonzero_leading_trailing_zeros", since = "1.53.0")]
```

Comment:
Just keep them `rustc_const_unstable`, then you don't have to stabilize `cttz_nonzero`.

Stabilizing `const` functions with UB such as `cttz_nonzero` is blocked on https://github.com/rust-lang/rfcs/pull/3016.

Reply:
ok was thinking that it was possible to make cttz_nonzero const stable due to ctlz_nonzero already was const stable.
shall the tracking issue still be the same and then not close it until trailing_zeros can be const stable?

Reply:
made a new tracking issue to make them const fn as there needed to be a new feature name as it not possible to have a feature that is stable and unstable.

Reply:
> ok was thinking that it was possible to make cttz_nonzero const stable due to ctlz_nonzero already was const stable.

Hm, good point -- I didn't realize that. We can probably also stabilize cttz_nonzero then. Cc @rust-lang/wg-const-eval 

However, neither of them have tests to ensure that UB due to passing "0" is properly detected. Could you add such a test? `src/test/ui/consts/const-int-unchecked.rs` seems like a reasonable file to add this to.

Reply:
I can try to make some test for passing '0' can do that as part of https://github.com/rust-lang/rust/issues/84089 so we do not hinder this stabilization.

Reply:
or do you think that I shall close #84089 and do all in this PR?

Reply:
> or do you think that I shall close #84089 and do all in this PR?

Whatever you prefer. I did not realize that we already have "intrinsics with UB" const-stabilized -- I guess this is okay since those intrinsics cannot actually be called directly from stable code, and all the functions that *can* be called are indeed safe.

Reply:
@andjo403 Let's just do it all in this PR. `ctlz_nonzero` is already const_stable, and the fcp on #79143 already included the constness of this function.

Reply:
added tests and did all in this PR so closed #84089